core/src/main/java/org/elasticsearch/index/query/QueryStringQueryParser.java

```diff
@@ -100,148 +100,109 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
-                    qpSettings.fuzzyMaxExpansions(parser.intValue());
+                    fuzzyMaxExpansions = parser.intValue();
                 } else if ("fuzzy_rewrite".equals(currentFieldName) || "fuzzyRewrite".equals(currentFieldName)) {
-                    qpSettings.fuzzyRewriteMethod(QueryParsers.parseRewriteMethod(parseContext.parseFieldMatcher(), parser.textOrNull()));
+                    fuzzyRewrite = parser.textOrNull();
                 } else if ("phrase_slop".equals(currentFieldName) || "phraseSlop".equals(currentFieldName)) {
-                    qpSettings.phraseSlop(parser.intValue());
+                    phraseSlop = parser.intValue();
                 } else if (parseContext.parseFieldMatcher().match(currentFieldName, FUZZINESS)) {
-                    qpSettings.setFuzziness(Fuzziness.parse(parser));
+                    fuzziness = Fuzziness.parse(parser);
                 } else if ("boost".equals(currentFieldName)) {
-                    qpSettings.boost(parser.floatValue());
+                    boost = parser.floatValue();
                 } else if ("tie_breaker".equals(currentFieldName) || "tieBreaker".equals(currentFieldName)) {
-                    qpSettings.tieBreaker(parser.floatValue());
+                    tieBreaker = parser.floatValue();
                 } else if ("analyze_wildcard".equals(currentFieldName) || "analyzeWildcard".equals(currentFieldName)) {
-                    qpSettings.analyzeWildcard(parser.booleanValue());
+                    analyzeWildcard = parser.booleanValue();
                 } else if ("rewrite".equals(currentFieldName)) {
-                    qpSettings.rewriteMethod(QueryParsers.parseRewriteMethod(parseContext.parseFieldMatcher(), parser.textOrNull()));
+                    rewrite = parser.textOrNull();
                 } else if ("minimum_should_match".equals(currentFieldName) || "minimumShouldMatch".equals(currentFieldName)) {
-                    qpSettings.minimumShouldMatch(parser.textOrNull());
+                    minimumShouldMatch = parser.textOrNull();
                 } else if ("quote_field_suffix".equals(currentFieldName) || "quoteFieldSuffix".equals(currentFieldName)) {
-                    qpSettings.quoteFieldSuffix(parser.textOrNull());
+                    quoteFieldSuffix = parser.textOrNull();
                 } else if ("lenient".equalsIgnoreCase(currentFieldName)) {
-                    qpSettings.lenient(parser.booleanValue());
+                    lenient = parser.booleanValue();
                 } else if ("locale".equals(currentFieldName)) {
                     String localeStr = parser.text();
-                    qpSettings.locale(LocaleUtils.parse(localeStr));
-                } else if ("time_zone".equals(currentFieldName)) {
+                    locale = Locale.forLanguageTag(localeStr);
+                } else if ("time_zone".equals(currentFieldName) || "timeZone".equals(currentFieldName)) {
```

Comment:
ok, fair enough. Maybe there should be an issue for this if there isn't already?